source/module_base/test/memory_test.cpp

```diff
@@ -0,0 +1,126 @@
+#include "../memory.h"
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include <fstream>
+#include <cstdio>
+
+/************************************************
+ *  unit test of class Memory
+ ***********************************************/
+
+/**
+ * - Tested Functions:
+ *   - calculate_mem
+ *     - caculate memory consumed for various types
+ *     - of data
+ *   - record
+ *     - record memory consumed during running
+ *   - print_all
+ *     - print memory consumed (> MB) in a
+ *     - std::ofstream file
+ */
+
+class MemoryTest : public testing::Test
+{
+protected:
+	// definition according to ../memory.cpp
+	double factor = 1.0 / 1024.0 / 1024.0; // MB
+	double complex_matrix_mem = 2*sizeof(double) * factor; // byte to MB
+	double double_mem = sizeof(double) * factor;
+	double int_mem = sizeof(int) * factor;
+	double bool_mem = sizeof(bool) * factor;
+	double float_mem = sizeof(float) * factor;
+	double short_mem = sizeof(short) * factor;
+	int n = 1024;
+	// for capturing stdout
+	std::string output;
+	// for output in file
+	std::ofstream ofs;
+	std::ifstream ifs;
+	void TearDown()
+	{
+		remove("tmp");
+	}
+};
+
+TEST_F(MemoryTest, calculatemem)
```

Comment:
`CalculateMem` would be better.

Reply:
OK, agree. I will change it in due course.